modal/image.py

```diff
@@ -1441,6 +1441,11 @@ def uv_sync(
         The `pyproject.toml` and `uv.lock` in `uv_project_dir` are automatically added to the build context. The
         `uv_project_dir` is relative to the current working directory of where `modal` is called.
 
+        NOTE: This does *not* install the project or workspace itself into the environment (this is equivalent to the
+        `--no-install-workspace` flag in the `uv sync` command) and you would be expected to add any local python source
+        files using `Image.add_local_python_source` or similar methods after this call. This ensures that updates to
+        your code wouldn't invalidate the uv_sync image cache.
```

Comment:
Should we actually document that "workspaces" are not supported _at all_ (currently it will error if they are defined).

Also FWIW the final sentence is maybe not a super compelling rationale. Could you rephrase to make it more about, like, "won't require reinstalling all third-party dependencies"?

(That said, this feels more like a limitation to our implementation / consequences of using this in a Modal Image build context, because it's not something you'd need to worry about when using `uv sync` locally.).